drgn_tools/iscsi.py

```diff
@@ -126,7 +137,49 @@ def print_iscsi_hosts(prog: Program) -> None:
                 shost.host_blocked.counter.value_(),
                 shost.host_failed.value_(),
                 shost.shost_state.format_(type_name=False),
-                hex(__get_iscsi_session(shost).value_()),
+                hex(__get_iscsi_session(prog, shost).value_()),
             ]
         )
     print_table(output)
+
+
+def print_iscsi_report(prog) -> None:
+    """
+    Prints a comprehensive report including iscsi table and session stats. More info can be added later if needed.
+    """
+    print_iscsi_table(prog)
+    print()
+    output = {}
+
+    for session in for_each_iscsi_session(prog):
+        print("**********")
+        conn = session.leadconn
+
+        persistent_address = escape_ascii_string(
+            conn.persistent_address.string_()
+        )
+        persistent_port = int(conn.persistent_port)
+        output["Session"] = hex(session.value_())
+        output["SID"] = str(int(session.cls_session.sid))
+        output["Persistent Portal"] = f"{persistent_address}:{persistent_port}"
+        output["Iface Name"] = escape_ascii_string(session.ifacename.string_())
+        output["Initiatorname"] = escape_ascii_string(
+            session.initiatorname.string_()
+        )
+        output["Targetname"] = escape_ascii_string(
+            session.targetname.string_()
+        )
```

Comment:
the session and connection status are required. Also please dump all the scsi device attached to this session and disk status as well.